build 1.59.0 failing with setuptools error

Log:
```
Requirement already satisfied: pip in /usr/local/lib/python3.9/site-packages (23.0.1)
Collecting pip
  Downloading pip-23.2.1-py3-none-any.whl (2.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 3.1 MB/s eta 0:00:00
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 23.0.1
    Uninstalling pip-23.0.1:
      Successfully uninstalled pip-23.0.1
Successfully installed pip-23.2.1
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Collecting grpcio==1.59.0
  Downloading grpcio-1.59.0.tar.gz (24.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 24.8/24.8 MB 4.7 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [10 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-uz2gmmv7/grpcio_2dc429f0ccfc44a38678115[47](https://github.com/IBM/oss-ecosystem-grpc/actions/runs/6367010357/job/17286674710#step:4:48)22dabdb/setup.py", line 95, in <module>
          import commands
        File "/tmp/pip-install-uz2gmmv7/grpcio_2dc429f0ccfc44a3867811[54](https://github.com/IBM/oss-ecosystem-grpc/actions/runs/6367010357/job/17286674710#step:4:55)722dabdb/src/python/grpcio/commands.py", line 31, in <module>
          import support
        File "/tmp/pip-install-uz2gmmv7/grpcio_2dc429f0ccfc44a38[67](https://github.com/IBM/oss-ecosystem-grpc/actions/runs/6367010357/job/17286674710#step:4:68)81154[72](https://github.com/IBM/oss-ecosystem-grpc/actions/runs/6367010357/job/17286674710#step:4:73)2dabdb/src/python/grpcio/support.py", line 111, in <module>
          errors.CompileError: diagnose_compile_error,
      AttributeError: module 'setuptools.errors' has no attribute 'CompileError'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/build_grpc.yaml
+++ b/.github/workflows/build_grpc.yaml
@@ -32,6 +32,7 @@ jobs:
           -w "/build" ppc64le/python:3.9 /bin/bash -ec '
           export GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=1;
           pip install --upgrade pip;
+          pip install --upgrade setuptools;
           pip install grpcio==${{ inputs.release }};
           WHEEL_PATH=$(find /root/.cache/pip/wheels/ -name "grpcio*.whl");
           cp $WHEEL_PATH /build/;'          

